virtme/commands/run.py

```diff
@@ -1293,28 +1301,56 @@ def do_it() -> int:
     else:
         virtme_init_cmd = "virtme-init"
 
+    initsh = []
+    if args.systemd:
+        kernelargs.append("virtme_systemd=1")
```

Comment:
detail: if you only need this in the init script, maybe you can simply add "virtme.systemd", no need to have an env var, no?

```suggestion
        kernelargs.append("virtme.systemd")
```

See how other var are used, e.g. `virtme.snapd`.

Also, when the init script is started with systemd, does systemd not set specific env vars or create specific files? (or maybe `$$` / `getpid()` would not be 1)

It is more a detail, already a lot of things are being exported, but if that's not needed and simple to get the info elsewhere, maybe that would be easier to deal with? (not sure, up to you :) )